editor/src/components/inspector/sections/layout-section/layout-system-subsection/layout-system-controls.tsx

```diff
@@ -273,34 +288,123 @@ export const PaddingControl = React.memo(() => {
     )
   }, [])
 
+  useSelectorWithCallback(
+    Substores.selectedViews,
+    selectedViewsSelector,
+    () => setOveriddenMode(null),
+    'PaddingControl setOveriddenMode',
+  )
+
+  const allUnset = React.useMemo(() => {
+    return (
+      paddingTop.controlStatus === 'trivial-default' &&
+      paddingBottom.controlStatus === 'trivial-default' &&
+      paddingLeft.controlStatus === 'trivial-default' &&
+      paddingRight.controlStatus === 'trivial-default'
+    )
+  }, [
+    paddingTop.controlStatus,
+    paddingBottom.controlStatus,
+    paddingLeft.controlStatus,
+    paddingRight.controlStatus,
+  ])
+
+  const useShorthand = React.useMemo(() => {
+    return shorthand.controlStatus === 'simple' || allUnset
+  }, [allUnset, shorthand.controlStatus])
+
+  const splitContolGroups = React.useMemo(
+    () =>
+      aggregateGroups({
+        top: paddingTop,
+        left: paddingLeft,
+        right: paddingRight,
+        bottom: paddingBottom,
+      }),
+    [paddingBottom, paddingLeft, paddingRight, paddingTop],
+  )
+
+  const aggregates = React.useMemo(
+    () =>
+      getSplitControlValues(splitContolGroups, {
+        top: paddingTop,
+        left: paddingLeft,
+        right: paddingRight,
+        bottom: paddingBottom,
+      }),
+    [paddingBottom, paddingLeft, paddingRight, paddingTop, splitContolGroups],
+  )
+
+  const values = React.useMemo(
+    () =>
+      getSplitChainedNumberInputValues(splitContolGroups, {
+        top: paddingTop,
+        left: paddingLeft,
+        right: paddingRight,
+        bottom: paddingBottom,
+      }),
+    [splitContolGroups, paddingBottom, paddingLeft, paddingRight, paddingTop],
+  )
+
+  const eventHandler = React.useMemo(
+    () =>
+      longhandShorthandEventHandler(
+        'padding',
+        {
+          T: 'paddingTop',
+          R: 'paddingRight',
+          B: 'paddingBottom',
+          L: 'paddingLeft',
+        },
+        selectedViewsRef,
+        useShorthand,
+        aggregates,
+        allUnset,
+        dispatch,
+      ),
+    [aggregates, allUnset, dispatch, selectedViewsRef, useShorthand],
+  )
+
+  const mode = React.useMemo(() => {
+    const initialMode = getInitialMode(
+      aggregates.oneValue,
+      aggregates.horizontal,
+      aggregates.vertical,
+      areAllSidesSet(splitContolGroups.allSides),
+      PaddingControlDefaultMode,
+    )
+
+    return initialMode
+  }, [aggregates.horizontal, aggregates.oneValue, aggregates.vertical, splitContolGroups.allSides])
+
+  const isCmdPressedRef = useRefEditorState((store) => store.editor.keysPressed.cmd === true)
+
+  const [overriddenMode, setOveriddenMode] = React.useState<ControlMode | null>('per-direction')
```

Comment:
for consistency with the border radius control, probably worth moving the default `per-direction` to a const

Reply:
fixed in https://github.com/concrete-utopia/utopia/pull/3289/commits/fc0d91ec845583f9f6aaecb23aa22af01d995432